### PR TITLE
Improve performance by buffering I/O operations

### PR DESCRIPTION
**Pull request by ddoktorski**
- Added buffers to speed up reading sierra file and saving casm file
- Added `rust-toolchain.toml` (perhaps setting Rust version can be omitted on CI, but I am not sure)

### CODE REVIEW

**Review by maciektr — APPROVED** [review]
🦾

File: rust-toolchain.toml
Comment (maciektr) [inline rust-toolchain.toml:1]: Not sure why you need it, but leaving to the Foundry team :)

**Review by cptartur — APPROVED** [review]
🙏

File: rust-toolchain.toml
Comment (ddoktorski) [reply rust-toolchain.toml:1]: We use version 1.79 on [CI](https://github.com/software-mansion/universal-sierra-compiler/blob/master/.github/workflows/ci.yml#L19), and it's not possible to compile with newer versions. So having it is simply convenient. This is not related to the main change.

**Pull request merged**